src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSObject.cs

```diff
@@ -1,13 +1,149 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Runtime.CompilerServices;
 using System.Runtime.Versioning;
 
 namespace System.Runtime.InteropServices.JavaScript
 {
+    /// <summary>
+    /// Holds proxy of JavaScript object.
+    /// </summary>
+    /// <remarks>Proxies are relatively expensive object. Developers could manualy <see cref="Dispose()"/> them to save runtime resources.</remarks>
     [SupportedOSPlatform("browser")]
     public partial class JSObject : IDisposable
     {
-        // TODO after https://github.com/dotnet/runtime/issues/70133
+        /// <summary>
+        /// Returns true if the proxy was already disposed.
+        /// </summary>
+        public bool IsDisposed { get => _isDisposed; }
+
+        /// <summary>
+        /// Returns true when the object contains the property.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool HasProperty(string propertyName)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Returns <code>typeof()</code> of the property.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public string GetTypeOfProperty(string propertyName)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Returns value of the property as <see cref="bool"/> if the property exists, otherwise false.
```

Comment:
We should have a brief line in these comments that describes what happens if the value's type is incompatible

Reply:
I'm still not convinced the current behavior is best choice. We would throw exception on wrong type, but marshal null as false or zero. That's kind of consistent with JS native behavior.

Cons of current approach is that it's not consistent from if you think about `undefined` and `null` as different types.
Cons of throwing exception on `null` and `undefined` is that exceptions are very expensive.

Reply:
I like this behavior. Most of the time I would want "bool or default" and in those cases when I want to distinguish between special JS values, we have a longer way to do it (`HasProperty`, `GetTypeOfProperty`).

Reply:
I added remarks about throwing JSException on wrong type and also reference to HasProperty, GetTypeOfProperty